java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -271,6 +332,29 @@ class DatabaseImpl(
         }
     }
 
+    fun getPrimitiveValue(valueId: FieldValueId, typeId: TypeId): Any {
```

Comment:
Should this be private?  Also, it may make sense to pass the `SQLiteDatabase` object to these private helper functions, I'm not sure what the impact is of - say - starting a transaction with `writableDatabase` and, within that transaction: using `readableDatabase`

Reply:
Marked it `@VisibleForTesting`, and made helper methods accept a db arg like you suggest